numba/targets/slicing.py

```diff
@@ -207,3 +207,30 @@ def slice_step_impl(context, builder, typ, value):
         return sli.step
     else:
         return context.get_constant(types.intp, 1)
+
+
+@lower_builtin("slice.indices", types.SliceType, types.Integer)
+def slice_indices(context, builder, sig, args):
+    length = args[1]
+    sli = context.make_helper(builder, sig.args[0], args[0])
+
+    with builder.if_then(cgutils.is_neg_int(builder, length), likely=False):
+        context.call_conv.return_user_exc(
+            builder, ValueError,
+            ("length should not be negative",)
+        )
```

Comment:
Could I just wrap this in a `if utils.IS_PY3` block to deal with those python 2.7 test failures?

Reply:
Not sure, why does this raise on 2.7 but not on the other python versions?

Reply:
This doesn't raise an error in 2.7. It does in python3. In python 2.7:

```python
>>> slice(None).indices(-10)
(-10, -10, 1)
```

From the "What's new" for python 3.4:

> slice.indices() no longer produces an OverflowError for huge values. As a consequence of this fix, slice.indices() now raises a ValueError if given a negative length; previously it returned nonsense values ([bpo-14794](https://bugs.python.org/issue14794)).